vm-c/vm.cpp

```diff
@@ -0,0 +1,248 @@
+#include <gmp.h>
+#include <iostream>
+#include "proto/msg.pb.h"
+#include "runtime/header.h"
+#include "runtime/alloc.h"
+#include "vm.h"
+#include "semantics.h"
+#include "world.h"
+
+using namespace io::iohk::ethereum::extvm;
+
+std::vector<mpz_ptr> k_to_z(list* l) {
+  std::vector<mpz_ptr> result;
+  for (size_t i = 0; i < hook_LIST_size_long(l); i++) {
+     zinj* elem = (zinj*) hook_LIST_get_long(l, i);
+     result.push_back(elem->data);
+  }
+  return result;
+}
+
+std::vector<account *> k_to_acct(map* m) {
+  list l = hook_MAP_values(m);
+  std::vector<account *> result;
+  for (size_t i = 0; i < hook_LIST_size_long(&l); i++) {
+     account* elem = (account*) hook_LIST_get_long(&l, i);
+     result.push_back(elem);
+  }
+  return result;
+}
+
+std::vector<log *> k_to_log(list* l) {
+  std::vector<log*> result;
+  for (size_t i = 0; i < hook_LIST_size_long(l); i++) {
+     loginj* elem = (loginj*) hook_LIST_get_long(l, i);
+     result.push_back(elem->data);
+  }
+  return result;
+}
+
+void k_to_log(log* log, LogEntry *pb) {
```

Comment:
Why isn't this one returning a freshly-allocated LogEntry?

Reply:
Because of how we allocate protobuf messages, the methods that populate a protobuf are taking a protobuf message as input and populating them. It could probably be rewritten, but then we would have to pass a pointer to the parent protobuf message, which doesn't really seem better to me.